### DISCUSSION TITLE
Customise elFinder down to only provide file browser and upload

### DISCUSSION THREAD

**Discussion by smileBeda** [Studio-42/elFinder#3481] [Q&A]
Hi, before I go and do the huge work I thought to ask if there's easier or quicker ways.

I want to use elFinder with CKEditor, but I do only need the file browsing feature, and the upload feature.
I do not need anything else, like:
- edit files
- mount drives
- integrations for gDrive, DropBox, etc
- trash
- etc

Is there some "easy" way to have a very stripped down version of elFinder? The current version is cool, but it is almost like a standalone CMS, and it is huge to maintain (security issues, libraries, etc etc all require a lot of attention)

I have begun to strip it down locally but was wondering if there is like some "light" version of the elFinder or processes we can use to strip out features we do not need.

**Comment by missenc** [comment]
Not an answer.  Just wondering how much progress you've made.  I'm attempting to do something similar and, while I have elFinder and CKEditor4 working well together, I cannot use CKEdit's hyperlink tool.  missenc@chipster.org

**Reply by smileBeda** [reply]
I have implemented a Custom File Browser without the elFinder because elFinder while certainly cool and very powerful, is a pandora box of problematic code, I found. Too much to maintain for a team, let alone one single person, and a quick scan revealed techniques I cannot feel conformable with nor deliver to end users as "vetted and safe".

Have a look at https://github.com/uhleloX/uhleloX. It a new CMS I am working on, and what I needed a file browser for.

The file browser I came up with is mostly in this extension https://github.com/uhleloX/uhleloX/tree/main/var/extensions/x-media-browser and it interacts with an AJAX operation sent to https://github.com/uhleloX/uhleloX/blob/main/ajax.php which then routes the data into specific methods that deliver response.
I took the same approach for the _upload_ feature, also seen in the extensions (x-ck-editor)

Feel free to jot any questions in that project, happy to help where I can.
